src/main/java/org/opensearch/ad/rest/handler/InsightsJobActionHandler.java

```diff
@@ -0,0 +1,388 @@
+                        Job existingJob = Job.parse(parser);
+
+                        if (existingJob.isEnabled()) {
+                            logger.info("Insights job is already running");
+                            listener.onResponse(new InsightsJobResponse("Insights job is already running"));
+                            return;
+                        }
+
+                        // Use current user if provided, otherwise keep existing user (for BWC)
+                        Job enabledJob = new Job(
+                            existingJob.getName(),
+                            createSchedule(frequency),
+                            existingJob.getWindowDelay(),
+                            true,
+                            Instant.now(),
+                            null,
+                            Instant.now(),
+                            existingJob.getLockDurationSeconds(),
+                            user != null ? user : existingJob.getUser(),
+                            existingJob.getCustomResultIndexOrAlias(),
+                            existingJob.getAnalysisType()
+                        );
+
+                        indexJob(
+                            enabledJob,
+                            listener,
+                            String.format(Locale.ROOT, "Insights job restarted successfully with frequency: %s", frequency)
+                        );
+
+                    } catch (IOException e) {
+                        logger.error("Failed to parse existing insights job", e);
+                        listener
+                            .onFailure(
+                                new OpenSearchStatusException("Failed to parse existing insights job", RestStatus.INTERNAL_SERVER_ERROR)
+                            );
+                    }
+                } else {
+                    createNewJob(frequency, user, listener);
+                }
+            }, e -> {
+                logger.error("Failed to check for existing insights job", e);
+                listener.onFailure(e);
+            }));
+        }
+    }
+
+    /**
+     * Create a brand new insights job.
+     */
+    private void createNewJob(String frequency, User user, ActionListener<InsightsJobResponse> listener) {
+        try {
+            IntervalSchedule schedule = createSchedule(frequency);
+            long lockDurationSeconds = java.time.Duration.of(schedule.getInterval(), schedule.getUnit()).getSeconds() * 2;
+
+            IntervalTimeConfiguration windowDelay = new IntervalTimeConfiguration(0L, ChronoUnit.MINUTES);
+
+            Job job = new Job(
+                ADCommonName.INSIGHTS_JOB_NAME,
+                schedule,
+                windowDelay,
+                true,
+                Instant.now(),
+                null,
+                Instant.now(),
+                lockDurationSeconds,
+                user,
+                ADCommonName.INSIGHTS_RESULT_INDEX_ALIAS,
```

Comment:
Can you add this index in ADIndex and try to reference ADIndex.INSIGHTS.getIndexName()? This would be consistent with other indexes.